### PR TITLE
Scope the OpenShift install version listing by location

### PR DESCRIPTION
**Pull request by darthhexx**
### Which issue this PR addresses:
Implements the API changes to support location specific OpenShift install version listing for the az CLI. [WI](https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/14177970).

### What this PR does / why we need it:
Each region has a dedicated CosmoDB database for storing the installable OpenShift versions. Adding the location parameter to the request ensures the request is sent to the correct ARO-RP instance.

Another use case for the change to scoping to location is that is comes with a _subscription_ scope and that will provide the RP with the capability to return different supported OpenShift versions for specific subscriptions. For example, an internal CI subscription for testing nightly builds.

### Test plan for issue:
Existing unit and e2e tests.

### Is there any documentation that needs to be updated for this PR?
Once the API is finalised for release.

### CODE REVIEW

File: pkg/frontend/frontend.go
Comment (bennerv) [inline pkg/frontend/frontend.go:221]: How are we going to direct this call to the associated RP?  Can you link some MSFT documentation on how the call goes to the specific resource provider based on this information?  

otherwise, we'd have to provide every version available in all RP cosmosdb databases.

**Review by bennerv — COMMENTED** [review]
Question on this

File: pkg/frontend/frontend.go
Comment (bennerv) [reply pkg/frontend/frontend.go:221]: If we look at the `az vm list-skus` call it's making a request to this guy: https://management.azure.com/subscriptions/fe16a035-e540-4ab7-80d9-373fa9a3d6ae/providers/Microsoft.Compute/skus?api-version=2019-04-01

This has to be location agnostic.

Comment (bennerv) [reply pkg/frontend/frontend.go:221]: @darthhexx I think this is what we want to implement. https://eng.ms/docs/products/arm/api_contracts/skuupdatecontract

@hawkowl we might have to revert the openshift versions in cosmosdb.  

Note that I can't view the doc on the page but I requested access.

Comment (bennerv) [reply pkg/frontend/frontend.go:221]: It looks like AKS follows a similar pattern to what you used:
`/subscriptions/subId/providers/Microsoft.ContainerService/locations/eastus/orchestrators`

I'm curious if they have store the configuration in a common place where every resource provider queries, every resource provider has all information about all locations, or something else.  Discussing with Weinong.

Comment (bennerv) [reply pkg/frontend/frontend.go:221]: TIL: 
the `/locations` API endpoint appears to route directly to the RP within that location. This looks good to me actually.  Sorry about the confusion, I didn't know that it's how it would work.  

>The usage can also be based on specific location. And in that case the API would look like below –

> GET https://management.azure.com/subscriptions/{subscriptionId}/providers/{providerNameSpace}/locations/{location}/usages?api-version={version}

**Comment by hawkowl** [comment]
I think this is good, just waiting for master to be unbroke before I land more.

**Review by hawkowl — APPROVED** [review]

**Pull request merged**
